spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -197,6 +203,22 @@ trait RewriteRowLevelOperationHelper extends PredicateHelper with Logging {
       }
     }
   }
+
+  protected def buildWritePlan(childPlan: LogicalPlan, table: Table): LogicalPlan = {
```

Comment:
Copied as it was.

Reply:
For a follow up, should we add a rule that does this for Append and Overwrite plans as well? That would be nice so that we don't have to wait until 3.2.0 to get write distribution and ordering for normal writes.

Reply:
I think we may consider doing that too.